baccmod/base_fit_acceptance_map_creator.py

```diff
@@ -0,0 +1,205 @@
+            max_fraction_pixel_rotation_fov=max_fraction_pixel_rotation_fov,
+            time_resolution=time_resolution,
+            use_mini_irf_computation=use_mini_irf_computation,
+            mini_irf_time_resolution=mini_irf_time_resolution,
+            interpolation_type=interpolation_type,
+            activate_interpolation_cleaning=activate_interpolation_cleaning,
+            interpolation_cleaning_energy_relative_threshold=interpolation_cleaning_energy_relative_threshold,
+            interpolation_cleaning_spatial_relative_threshold=interpolation_cleaning_spatial_relative_threshold,
+        )
+
+        self.sq_rel_residuals = {"mean": [], "std": []}
+        self.list_name_normalisation_parameter = list_name_normalisation_parameter
+
+    @abstractmethod
+    def create_acceptance_map(self, observations):
+        """
+        Subclasses must implement: run self._create_base_computation_map(), then
+        call self.fit_background() for each energy slice, and finally pack into Background3D.
+        """
+        pass
+
+    def fit_background(self, model: Model, *coords: List[np.ndarray], count_map: np.ndarray, exp_map_total: np.ndarray, exp_map: np.ndarray) -> np.ndarray:
+        """
+        Perform a Poisson fit on the given map (could be 1D, 2D or 3D), given
+        the fine‐binned exposure (exp_map) and total‐exposure (exp_map_total),
+        returning a smooth “counts model” on the same grid as `count_map`.
+
+        Parameters
+        ----------
+        model : astropy.modeling.model
+            the model to fit to the data
+        coords : List of np.ndarray
+            the list of coordinate
+        count_map : np.ndarray
+            counts in each pixel, integer
+        exp_map_total : np.ndarray
+            exposure WITHOUT exclusion regions
+        exp_map : np.ndarray
+            exposure CORRECTED for exclusion regions
+
+        Returns
+        -------
+        fitted_counts_model : 2D np.ndarray
+            The best‐fit model of counts (on the same pixel grid as count_map).
+        """
+
+        # Compute exposure correction
+        mask = exp_map_total > 0
+        exp_map_total_safe = exp_map_total.copy()
+        exp_map_total_safe[mask] = 1
+        exp_correction = exp_map / exp_map_total_safe
+
+        # Initialise the model
+        model_init = model.copy()
+
+        # Correct normalisation of the model
+        if self.list_name_normalisation_parameter is not None and len(self.list_name_normalisation_parameter) > 0:
+            # Compute correction
+            init_count_model = np.sum(model_init(*coords)*exp_correction)
+            correction_norm = np.sum(count_map)/init_count_model
+
+            # build list of free parameters
+            tied = model_init.tied
+            all_params = list(model_init.param_names)
+            indep_params = [p for p in all_params if (p not in tied) or (not tied[p]) and (not model_init.fixed.get(p, False))]
```

Comment:
Why remove the frozen parameters from the normalisation code? You could want to force the amplitude to be normalised to the observed counts and not fitted.